Task5_1_SparseMatrix.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task5_1
+{
+    class SparseMatrix : IEnumerable<long>
+    {
+        private readonly int rows;
+        private readonly int columns;
+        private readonly Dictionary<int, Dictionary<int, long>> matrixData;
+
+        public SparseMatrix(int rows, int columns)
+        {
+            if (rows <= 0 || columns <= 0)
+                throw new ArgumentException("Rows and columns must be strictly greater than zero.");
+
+            this.rows = rows;
+            this.columns = columns;
+            this.matrixData = new Dictionary<int, Dictionary<int, long>>();
+        }
+
+        public long this[int i, int j]
+        {
+            get
+            {
+                if (i < 0 || i >= rows || j < 0 || j >= columns)
+                    throw new IndexOutOfRangeException("Invalid matrix indices.");
+
+                return matrixData.ContainsKey(i) && matrixData[i].ContainsKey(j) ? matrixData[i][j] : 0;
+            }
+            set
+            {
```

Comment:
Better implement a separate method for indexes range check, do not repeat yourself  (DRY)

Reply:
corrected